packages/grid/x-data-grid/src/constants/gridClasses.ts

```diff
@@ -574,6 +578,7 @@ export const gridClasses = generateUtilityClasses<GridClassKey>('MuiDataGrid', [
   'row--lastVisible',
   'row--dragging',
   'row--dynamicHeight',
+  'row--expanded',
```

Comment:
Maybe it should have more specific naming like `row--detailPanelExpanded`?
Row can also be "expanded" when using row grouping feature